Provider.php

```diff
@@ -69,11 +69,41 @@ public function getLastTouched(array $tableNamesOrIds)
                 array_merge($ids, $names)
             );
 
-            return $timestamp;
+            if ($lastTouched === null) {
+                return null;
+            }
+
+            $lastTouchedObject = new \DateTime($lastTouched);
+            return $lastTouchedObject->getTimestamp();
         }
     }
 
     /**
+     * Returns all tracked data rows and their respective last changes of a given table.
+     *
+     * @param string $tableName
+     * @return array
+     */
+    public function getLastTouchedOfEachRow($tableName)
```

Comment:
In other places, a table id can be passed as well, but since you don't need it, we can add that later on.